source/lac/trilinos_solver.cc

```diff
@@ -480,6 +486,81 @@ namespace TrilinosWrappers
     verbose_cout << "Starting numeric factorization" << std::endl;
     ierr = solver->NumericFactorization();
     AssertThrow (ierr == 0, ExcTrilinosError(ierr));
+  }
+
+
+  void SolverDirect::solve (VectorBase &x, const VectorBase &b)
+  {
+    // Assign the empty LHS vector to the Epetra_LinearProblem object
+    linear_problem->SetLHS(&x.trilinos_vector());
+
+    // Assign the RHS vector to the Epetra_LinearProblem object
+    linear_problem->SetRHS(const_cast<Epetra_MultiVector *>(&b.trilinos_vector()));
+
+    // Fetch return value of Amesos Solver functions
+    int ierr;
+
+    // First set whether we want to print the solver information to screen or
+    // not.
+    ConditionalOStream verbose_cout (std::cout,
+                                     additional_data.output_solver_details);
+
+
+    verbose_cout << "Starting solve" << std::endl;
+    ierr = solver->Solve ();
+    AssertThrow (ierr == 0, ExcTrilinosError (ierr));
+
+    // Finally, let the deal.II SolverControl object know what has
```

Comment:
I think this comment should be changed as it was probably copy-pasted from iterative solver. Here you just make `solver_control` happy by providing 0 residual so that it is bound to succeed. One could still calculate residual to be sure that it worked?